src/lib/frontend/typechecker.ml

```diff
@@ -2143,9 +2144,30 @@ let type_user_defined_type_body ~is_recursive env acc (loc, ls, s, body) =
     assert (not is_recursive); (* Abstract types are not recursive *)
     acc, env
 
-let rec type_decl (acc, env) d =
+let rec push_pop_env f n s =
+  if n <= 1 then
+    f s
+  else
+    let _ = f s in
+    push_pop_env f (n-1) s
+
+let rec type_decl (acc, env) d assertion_stack =
   Types.to_tyvars := MString.empty;
   match d with
+  | Push (loc,n) ->
+    push_pop_env (Stack.push env) n assertion_stack;
```

Comment:
Should we check that `n >= 0` or ` n >= 1` ?

Reply:
I have added a check for `n >= 0` in the typechecker and in psmt2_to_alt_ergo.ml since the smtlib standard only spécify that n should be `>= 0`